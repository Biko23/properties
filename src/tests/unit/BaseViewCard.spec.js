import 'jsdom-global'
import { createLocalVue, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import BaseViewCard from '../../components/BaseViewedCard.vue'

describe('BaseViewCard.vue', () => {
  const localVue = createLocalVue()
    let vuetify

    beforeEach(() => {
        vuetify = new Vuetify()
    })

    const mountFunction = options => {
        return mount(BaseViewCard, {
          localVue,
          vuetify,
          ...options
        })
    }

    it('should be a function', () => {
        const wrapper = mountFunction()
        expect(typeof wrapper.data).toBe('function')
    })

//   it('should have a custom date and match snapshot', () => {
//     const wrapper = mountFunction({
//         propsData: { date: '13th/12/2021', cost: '99738' }
//       })
  
//       expect(wrapper.html()).toMatchSnapshot()

//       const category = wrapper.find('.test-subtitle  small')
//       expect(category.text()).toEqual('UGX: 99738')
//   })
})

