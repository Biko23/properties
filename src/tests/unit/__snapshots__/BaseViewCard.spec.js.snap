// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseViewCard.vue should have a custom date and match snapshot 1`] = `
"<div class=\\"v-card v-sheet theme--light\\" style=\\"max-width: auto;\\">
  <div class=\\"v-card__subtitle time\\" style=\\"z-index: 1;\\">Viewed 13th/12/2021 </div>
  <div class=\\"v-image v-responsive theme--light\\" style=\\"height: 200px;\\">
    <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
      <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
    </transition-stub>
    <div class=\\"v-responsive__content\\"></div>
  </div>
  <div style=\\"display: flex; flex-direction: row; justify-content: space-between;\\">
    <div class=\\"v-card__subtitle test-subtitle\\"><small style=\\"font-weight: bold; color: rgb(0, 0, 0); margin-bottom: 0px;\\">UGX: 99738</small><br> <small>Category: Category missing</small><br> <small>Located: No available location</small><br> <small>Posted by: No name</small></div>
  </div>
</div>"
`;
